src/main/kotlin/org/team2471/frc/pathvisualizer/ControlPanel.kt

```diff
@@ -333,19 +334,21 @@ object ControlPanel : VBox() {
 
         val playButton = Button(" Play ")
         playButton.setOnAction {
-            if (selectedPath!=null) {
+            if (selectedPath != null) {
                 val timer = Timer()
                 timer.start()
                 thread {
                     while (timer.get() < selectedPath!!.durationWithSpeed) {
-                        currentTime = timer.get()
-                  //      draw()
-                  //      refresh()
-                   //     println("time: $currentTime")
+                        Platform.runLater {
+                            currentTime = timer.get()
+                            draw()
+                            refresh()
+                        }
+
+                        // Playback @ approx 30fps (1000ms/30fps = 33ms)
+                        Thread.sleep(1000L / 30L)
```

Comment:
I had to limit this and add a delay here because otherwise we would be swamping the main thread with too many jobs. This was as quick as I could make it without causing hiccups in the playback. There might be a better way to do this, but I wasn't able to find anything.